corehq/apps/api/resources/v0_5.py

```diff
@@ -1044,3 +1051,32 @@ def prepend_urls(self):
             url(r"^(?P<resource_name>{})/(?P<config_id>[\w\d_.-]+)/feed".format(
                 self._meta.resource_name), self.wrap_view('dispatch_list')),
         ]
+
+
+class NavigationEventAuditResource():
+
+    @classmethod
+    def query(cls, domain: str, local_time_zone: ZoneInfo, users: list = [],
+            local_start_date: date = None, local_end_date: date = None):
+        filters = Q(domain=domain)
+        if users:
+            filters &= Q(user__in=users)
+
+        queryset = NavigationEventAudit.objects.filter(filters)
+
+        results = (queryset
+                .annotate(local_date=TruncDate('event_date', tzinfo=local_time_zone))
+                .values("local_date", "user")
+                .annotate(UTC_first_action_time=Min('event_date'), UTC_last_action_time=Max('event_date')))
+
+        date_filters = Q()
+        if local_start_date:
+            date_filters &= Q(local_date__gte=local_start_date)
+        if local_end_date:
+            date_filters &= Q(local_date__lte=local_end_date)
+
+        results = results.filter(date_filters).order_by('local_date', 'user')
```

Comment:
Thinking of performance and pulling a smaller set of results, is there a way to apply date filters first if present? Or does it even matter? (I know there's often some behind-the-scenes query optimization)

Reply:
Yes, I think the order does matter and filtering first is better for performance. I did end up doing date_filter before applying `annotate(UTC_first_action_time=Min('event_date'), UTC_last_action_time=Max('event_date')` (haven't committed yet). 

The filter has to go after the `local_date ` annotation but you have a good point, I think the filtering can be done before the `values` clause